src/settings.cpp

```diff
@@ -991,30 +924,24 @@ static bool TrainAccelerationModelChanged(int32 p1)
 	SetWindowClassesDirty(WC_ENGINE_PREVIEW);
 	InvalidateWindowClassesData(WC_BUILD_VEHICLE, 0);
 	SetWindowClassesDirty(WC_VEHICLE_DETAILS);
-
-	return true;
 }
 
 /**
  * This function updates the train acceleration cache after a steepness change.
- * @param p1 Callback parameter.
- * @return Always true.
+ * @param new_value Callback parameter.
```

Comment:
Not sure if you want to do this in this PR, but doxygen comments on these callbacks are a bit touch-and-go. Some don't have them, some don't mention `new_value`, and some do.

Either way, maybe also fix `Callback parameter` to be a bit more descriptive like `New value of the setting` or what-ever :)